### DISCUSSION TITLE
Quick link retrieval from many tabs

### DISCUSSION THREAD

**Discussion by veyorjoachim** [cssnr/link-extractor#59] [Feature Requests]
I have several open tabs and I want to quickly retrieve urls with regex from all tabs into one single txt. Is it possible to add a keyboard shortcut (let's say ctrl+Q or example) so that I can quickly shift between tabs and hit the shortcut and Link Extractor will append relevant urls to the txt? Then I could hit ctrl+TAB and ctrl+Q repeatedly and at the end I download my desired list of urls.

**Comment by smashedr** [comment]
Sorry it took so long to implement this feature. I was waiting for Permissions Handling to improve in Manifest Version 3.

I have added an Optional Host Permissions Flow, which once granted, allows you to extract links from ALL tabs you have selected.

![20240509-004907269](https://github.com/cssnr/link-extractor/assets/6071159/0cdb2190-8d00-455c-ba7f-a075f289c8b3)
![20240509-004915753](https://github.com/cssnr/link-extractor/assets/6071159/004d8e60-7200-4687-90b4-10837c4200b1)

So just use Ctrl+Click to select as many tabs as you want, then you can extract via the popup from all the selected tabs.

Let me know if you have any questions or issues with the feature.

Cheers.
